packages/core/src/shared/clients/apprunner.ts

```diff
@@ -0,0 +1,142 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+    AppRunnerClient as AppRunnerClientSDK,
+    CodeConfiguration,
+    CodeConfigurationValues,
+    CodeRepository,
+    ConnectionSummary,
+    CreateConnectionCommand,
+    CreateConnectionRequest,
+    CreateConnectionResponse,
+    CreateServiceCommand,
+    CreateServiceRequest,
+    CreateServiceResponse,
+    DeleteServiceCommand,
+    DeleteServiceRequest,
+    DeleteServiceResponse,
+    DescribeServiceCommand,
+    DescribeServiceRequest,
+    DescribeServiceResponse,
+    ImageRepository,
+    ListConnectionsCommand,
+    ListConnectionsRequest,
+    ListConnectionsResponse,
+    ListOperationsCommand,
+    ListOperationsRequest,
+    ListOperationsResponse,
+    ListServicesCommand,
+    ListServicesRequest,
+    ListServicesResponse,
+    paginateListServices,
+    PauseServiceCommand,
+    PauseServiceRequest,
+    PauseServiceResponse,
+    ResumeServiceCommand,
+    ResumeServiceRequest,
+    ResumeServiceResponse,
+    ServiceSummary,
+    SourceCodeVersion,
+    SourceConfiguration,
+    StartDeploymentCommand,
+    StartDeploymentRequest,
+    StartDeploymentResponse,
+    UpdateServiceCommand,
+    UpdateServiceRequest,
+    UpdateServiceResponse,
+} from '@aws-sdk/client-apprunner'
+import { ClientWrapper } from './clientWrapper'
+import { hasProps, RequiredProps } from '../utilities/tsUtils'
+import { AsyncCollection } from '../utilities/asyncCollection'
+
+export type AppRunnerServiceSummary = RequiredProps<
+    ServiceSummary,
+    'ServiceName' | 'ServiceArn' | 'Status' | 'ServiceId'
+>
+export type AppRunnerImageRepository = RequiredProps<ImageRepository, 'ImageIdentifier' | 'ImageRepositoryType'>
+
+export type AppRunnerCodeConfigurationValues = RequiredProps<CodeConfigurationValues, 'Runtime'>
+interface AppRunnerCodeConfiguration extends RequiredProps<CodeConfiguration, 'ConfigurationSource'> {
+    CodeConfigurationValues: AppRunnerCodeConfigurationValues
+}
+export interface AppRunnerCodeRepository extends RequiredProps<CodeRepository, 'RepositoryUrl'> {
+    SourceCodeVersion: RequiredProps<SourceCodeVersion, 'Type' | 'Value'>
+    CodeConfiguration: AppRunnerCodeConfiguration
+}
+export interface AppRunnerSourceConfiguration extends SourceConfiguration {
```

Comment:
I would not bother with the `AppRunnerSource` prefix. There would be no conflict if we did it like this:

```suggestion
export interface SourceConfiguration extends AppRunner.SourceConfiguration {
```

Reply:
I agree its not a great solution. One alternative would be to shadow the names of the underlying types, and import them as different names. Ex. 
```
import { SourceConfiguration as SourceConfigurationSDK } from '@aws-sdk/client-apprunner'
...
export interface SourceConfiguration extends SourceConfigurationSDK {
```
My main reason for not doing this is that it could create confusion in cases where the custom type doesn't completely mirror the sdk type. As an example, in EC2 we add a `LastSeenStatus` field to the wrapping type that doesn't exist on the sdk type.

Reply:
> it could create confusion in cases where the custom type doesn't completely mirror the sdk type.

The types are coming from this module, thus they aren't the same. They have the same names, but developers are expected to understand that any module can use conflicting names anywhere. If one imports `Foo` from module A, that's unrelated to `Foo` from module B, even though they have the same name.

Reply:
That makes sense, and seems in-line with how we construct the clients. I'll adjust the names of the types in this PR and look into a followup to refactor the existing work done.